src/radical/pilot/utils/component.py

```diff
@@ -361,6 +371,11 @@ def declare_publisher(self, topic, pubsub):
         self._log.debug('declared publisher : %s : %s : %s' \
                 % (topic, pubsub, q.name))
 
+        # FIXME: I am not exactly sure why this is 'needed', but declaring
+        #        a published as above and then immediately publishing on that
+        #        channel seems to sometimes lead to a loss of messages.
+        time.sleep(1)
+
```

Comment:
Yeah, me too.  Its a `FIXME` because I very much would like to understand it.  If you feel strong about it, I'll dig into it before merge -- but I am rather confident that no messages get lost at this point.